server/game/cards/03_TWI/units/PadawanStarFighter.ts

```diff
@@ -0,0 +1,23 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { Trait } from '../../../core/Constants';
+
+export default class PadawanStarFighter extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '4718895864',
+            internalName: 'padawan-starfighter'
+        };
+    }
+
+    protected override setupCardAbilities() {
+        this.addConstantAbility({
+            title: 'While you control a Force unit or a Force upgrade, this unit gets +1/+1',
+            condition: (context) => context.source.controller.getOtherUnitsInPlayWithTrait(context.source, Trait.Force).length > 0 || context.source.controller.getArenaCards().filter((c) => c.isUpgrade)
```

Comment:
I think you can use isTraitInPlay(Trait.Force, context.source) instead of getOtherUnitInPlayWithTrait().length>0

Reply:
This a good use case for the new filter methods (still partially WIP) - I think `context.source.controller.getArenaCards({ trait: Force })` would grab everything

Reply:
Thanks for the pointer. Knowing what are the existing helper to check condition is the part that took me most of my time.